osquery/tables/system/darwin/mdls.mm

```diff
@@ -60,25 +61,56 @@
 
   return rvalue;
 }
+
 void genResults(const std::string& path, QueryData& results) {
   CFStringRef cs = CFStringCreateWithCString(
       kCFAllocatorDefault, path.c_str(), kCFStringEncodingASCII);
+  auto const cs_guard = scope_guard::create([cs]() { CFRelease(cs); });
```

Comment:
These statements need to be swapped as `CFRelease` will be called with `nullptr` (and crash the program) if the allocation function returns null. 

Is it possible to use `cs` by reference in the lambda to save an allocation?

Same concerns below.